tests/unittests/test_common.h

```diff
@@ -0,0 +1,476 @@
+/*
+Copyright (c) 2020-Present Advanced Micro Devices, Inc. All rights reserved.
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+/*
+ * File is intended to C and CPP compliant hence any CPP specic changes
+ * should be added into CPP section
+ *
+ */
+
+#ifdef __cplusplus
+    #include <iostream>
+    #include <iomanip>
+    #if __CUDACC__
+        #include <sys/time.h>
+    #else
+        #include <chrono>
+    #endif
+#endif
+
+// ************************ GCC section **************************
+#include <stddef.h>
+
+#include "hip/hip_runtime.h"
+#include "hip/hip_runtime_api.h"
+
+#define HC __attribute__((hc))
+
+
+#define KNRM "\x1B[0m"
+#define KRED "\x1B[31m"
+#define KGRN "\x1B[32m"
+#define KYEL "\x1B[33m"
+#define KBLU "\x1B[34m"
+#define KMAG "\x1B[35m"
+#define KCYN "\x1B[36m"
+#define KWHT "\x1B[37m"
+
+#define passed()                                                                                   \
+    printf("%sPASSED!%s\n", KGRN, KNRM);                                                           \
+    exit(0);
```

Comment:
should this be ```return true``` instead?

Reply:
Removed exit(0) as we will have return in main anyway.